schema.ts

```diff
@@ -404,9 +410,52 @@ export interface ClassMethod extends FunctionLike {
 }
 
 /**
- *
+ * A description of a mixin.
+ * 
+ * Mixins are functions which generate a new subclass of a given superclass.
+ * This interfaces describes the class and custom element features that
+ * are added by the mixin. As such, it extends the CustomElement interface and
+ * ClassLike interface.
+ * 
+ * Since mixins are functions, it also extends the FunctionLike interface. This
+ * means a mixin is callable, and has parameters and a return type.
+ * 
+ * The return type is often hard or impossible to accurately describe in type
+ * systems like TypeScript. It requires generics and an `exrtends` operator
+ * that TypeScript lacks. Therefore it's reccomended that the return type is
+ * left empty. The most common form of a mixin function takes a single
+ * argument, so consumers of this interface should assume that the return type
+ * is the single argument subclassed by this declaration.
+ * 
+ * @example
+ * 
+ * This JavaScript mixin declaration:
+ * ```javascript
+ * const MyMixin = (base) => class extends base {
+ *   foo() { ... }
+ * }
+ * ```
+ * 
+ * Is described by this JSON:
+ * ```json
```

Comment:
Should it still have a `"superclass"` field that has as value `base` here? Thats how I currently handle it in `@custom-elements-manifest/analyzer`, but it might be redundant I guess?

Reply:
No - mixins (usually) don't have superclasses until they are applied to a specific superclass.

When mixins do have a superclass, it's because of mixin composition:

```js
const MixinA = (base) => class extends base {
  fieldA;
}

const MixinB = (base) class extends MixinA(base) {
  fieldB;
}

class C extends MixinB(S) {
}
const c = new C();
c.fieldA; // exists
c.fieldB; // exists
```

Reply:
I added a short mention of this in the docs

Reply:
When a mixin has nested mixins, though, are those nested mixins documented as being mixins, or superclasses? E.g.:

```js
const MixinA = (base) => class extends base {
  fieldA;
}

const MixinB = (base) => class extends MixinA(base) {
  fieldB;
}
```

Would the mixin declaration look like this:

```json
        {
          "kind": "mixin",
          "description": "",
          "name": "MixinB",
          "mixins": [
            {
              "name": "MixinA"
            }
          ],
         "parameters": [
             {
               "name": "base"
             }
           ],
          "members": ["//etc"]
        }
```

If we consider `MixinA` to be the superclass in that case, we have to change `superclass` to be `superclasses` and be an array, because there can be multiple mixins applied.

Intuitively, I feel like I'd prefer to name "nested mixins" just mixins in this case though

Reply:
That's a good point, I'll change the comments to says composed mixins should go in the mixins field.